File: examples/benchmarks/pi_leibniz.c
#include <stdio.h>
#include <time.h>

double calculate_pi(int iterations) {
    double pi = 0.0;
    int sign = 1;
    
    for (int k = 0; k < iterations; k++) {
        pi += sign / (2.0 * k + 1.0);
        sign = -sign;
    }
    
    return pi * 4.0;
}

int main() {
    int iterations = 100000000;
    
    clock_t start = clock();
    double pi = calculate_pi(iterations);
    clock_t end = clock();
    
    double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;
    
    printf("π ≈ %.10f\n", pi);
    printf("Iterations: %d\n", iterations);
    printf("Time: %.6f seconds\n", time_taken);
    
    return 0;
}

File: examples/benchmarks/pi_leibniz.rs
use std::time::Instant;

fn calculate_pi(iterations: usize) -> f64 {
    let mut pi = 0.0;
    let mut sign = 1.0;
    
    for k in 0..iterations {
        pi += sign / (2.0 * k as f64 + 1.0);
        sign = -sign;
    }
    
    pi * 4.0
}

fn main() {
    let iterations = 100_000_000;
    
    let start = Instant::now();
    let pi = calculate_pi(iterations);
    let duration = start.elapsed();
    
    println!("π ≈ {:.10}", pi);
    println!("Iterations: {}", iterations);
    println!("Time: {:.6} seconds", duration.as_secs_f64());
}

